python/lsst/faro/utils/stellarLocus.py

```diff
@@ -0,0 +1,232 @@
+        xBoxMax = paramDict["xMax"]
+
+    bPerpMin = yBoxMin - mPerp*xBoxMin
+
+    paramsOut["yBoxMin"] = yBoxMin
+    paramsOut["bPerpMin"] = bPerpMin
+
+    bPerpMax = yBoxMax - mPerp*xBoxMax
+
+    paramsOut["yBoxMax"] = yBoxMax
+    paramsOut["bPerpMax"] = bPerpMax
+
+    # Use these perpendicular lines to chose the data and refit
+    fitPoints = ((ys > mPerp*xs + bPerpMin) & (ys < mPerp*xs + bPerpMax))
+    data = scipyODR.Data(xs[fitPoints], ys[fitPoints])
+    odr = scipyODR.ODR(data, linear, beta0=[bODR, mODR])
+    params = odr.run()
+    mODR = float(params.beta[1])
+    bODR = float(params.beta[0])
+
+    paramsOut["mODR2"] = float(params.beta[1])
+    paramsOut["bODR2"] = float(params.beta[0])
+
+    paramsOut["mPerp"] = -1.0/paramsOut["mODR2"]
+
+    # Calculate the fit lines for the entire X, Y range
+    xMinLine = np.min(xs)-0.2
+    xMaxLine = np.max(xs)+0.2
+    yMinLine = np.min(ys)-0.2
+    yMaxLine = np.max(ys)+0.2
+
+    if np.fabs(paramsOut["mHW"]) > 1:
+        ysFitLineHW = np.array([yMinLine, yMaxLine])
+        xsFitLineHW = (ysFitLineHW - paramsOut["bHW"])/paramsOut["mHW"]
+        yFiducialHW = np.array(paramsOut['yMin'])
+        xFiducialHW = (yFiducialHW - paramsOut["bODR"])/paramsOut["mODR"]
+        ysFitLine = np.array([yMinLine, yMaxLine])
+        xsFitLine = (ysFitLine - paramsOut["bODR"])/paramsOut["mODR"]
+        ysFitLine2 = np.array([yMinLine, yMaxLine])
+        xsFitLine2 = (ysFitLine2 - paramsOut["bODR2"])/paramsOut["mODR2"]
+        yFiducial = np.array(paramsOut['yMin'])
+        xFiducial = (yFiducial - paramsOut["bODR2"])/paramsOut["mODR2"]
+    else:
+        xsFitLineHW = np.array([xMinLine, xMaxLine])
+        ysFitLineHW = paramsOut["mHW"]*xsFitLineHW + paramsOut["bHW"]
+        xFiducialHW = np.array(paramsOut['xMin'])
+        yFiducialHW = paramsOut["mHW"]*xFiducialHW + paramsOut["bHW"]
+        xsFitLine = np.array([xMinLine, xMaxLine])
+        ysFitLine = [paramsOut["mODR"]*xsFitLine[0] + paramsOut["bODR"],
+                     paramsOut["mODR"]*xsFitLine[1] + paramsOut["bODR"]]
+        xsFitLine2 = np.array([xMinLine, xMaxLine])
+        ysFitLine2 = [paramsOut["mODR2"]*xsFitLine2[0] + paramsOut["bODR2"],
+                      paramsOut["mODR2"]*xsFitLine2[1] + paramsOut["bODR2"]]
+        xFiducial = np.array(paramsOut['xMin'])
+        yFiducial = paramsOut["mODR2"]*xFiducial + paramsOut["bODR2"]
+
+    # Calculate the distances to the fit line
+    # Need two points to characterise the lines we want
+    # to get the distances to
+    pt1 = np.array([xFiducial, yFiducial])
+    pt2 = np.array([xsFitLine2[1], ysFitLine2[1]])
+
+    pt1HW = np.array([xFiducialHW, yFiducialHW])
+    pt2HW = np.array([xsFitLine2[1], ysFitLineHW[1]])
+
+    # "p2" is the distance from the fit line, and "p1" is the distance
+    # along the line.
+    p1HW, p2HW = parallelPerpDistance(pt1HW, pt2HW, zip(xs, ys))
+    p1, p2 = parallelPerpDistance(pt1, pt2, zip(xs, ys))
```

Comment:
I think that you could use more descriptive names that both tell you what the variables are and don't look so much like pt1 and pt2.

Reply:
Maybe I should change "pt1" and "pt2", then. Because "p1" and "p2" are the specific names used in Ivezic+2004, so I'd like to keep those.